Controller/MediaAdminController.php

```diff
@@ -80,19 +80,19 @@ public function listAction(Request $request = null)
         $category = $this->container->get('sonata.classification.manager.category')->getRootCategory($context);
 
         if (!$filters) {
-            $datagrid->setValue('category', null, $category->getId());
+            $datagrid->setValue('category', null, $category);
         }
 
         if ($request->get('category')) {
-            $contextInCategory = $this->container->get('sonata.classification.manager.category')->findBy(array(
+            $categoryByContext = $this->container->get('sonata.classification.manager.category')->findOneBy(array(
```

Comment:
Please make separate commits for things you find in the way that do not relate to the PR (use `git add --patch`)